src/Zenaton/v1/Client.php

```diff
@@ -104,6 +107,23 @@ public function getWebsiteUrl($ressources = '', $params = '')
         return $this->addAppEnv($url, $params);
     }
 
+    /**
+     * Start a single task.
+     */
+    public function startTask(TaskInterface $task)
+    {
+        $response = $this->http->post($this->getTaskWorkerUrl(), [
+            self::ATTR_PROG => self::PROG,
+            self::ATTR_NAME => get_class($task),
+            self::ATTR_DATA => $this->serializer->encode($this->properties->getPropertiesFromObject($task)),
+            self::ATTR_MAX_PROCESSING_TIME => method_exists($task, 'getMaxProcessingTime') ? $task->getMaxProcessingTime() : null,
+        ]);
+
+        if ($response->code === 400) {
```

Comment:
I think you get a 400 if you aren't listening to the agent also. So in that case it seems that you'd have a strange error message. 

I think the only way to distinguish these two cases is to inspect the error message you get in the body of the response. In the case of bad version, you get 
```
{"error":"Unknown version. Please upgrade your worker"}
```
whereas when you haven't listened yet you get:
```
{"error":"Your worker does not listen to app \"APPID\" on env \"dev\""}
```

Reply:
Inspecting error messages is so much trouble ... I don't like that at all, but for now there is no real alternative. I might open an RFC because we need to find something better before we're inspecting many error messages in many different places.

For now I made the change as you suggested 👍